.github/workflows/test.yml

```diff
@@ -54,6 +54,12 @@ jobs:
         # to work with GNU grep.
         ! grep -qE "^.*?/src/[^:]+:[0-9]+: ?[a-zA-Z]+: ?.*$" doxygen-log.txt
 
+    - name: Expose archive docs artifact
```

Comment:
[question] Do you think it would be useful to have the artifact even if there are Doxygen errors/warnings, e.g. for debugging? 

By default the next step won't run if the previous step failed.  But that can be tweaked, e.g. see [stackoverflow answer](https://stackoverflow.com/a/58003436/535103) and ["steps" context docs](https://docs.github.com/en/actions/learn-github-actions/contexts#steps-context).

Perhaps though it's not worth the additional effort until we actually need it.

Reply:
I'd be happy to leave this as-is for now, but @steve-adamsATfoundry can we squash these two commits together please?